### PR TITLE
BD-1069: Add support for array of objects data type

### PR DESCRIPTION
**Pull request by bre-fitzgerald**
# Pull Request/Issue Resolution

Closes #**BD-1069**

#### Is this change associated with a Braze feature/product release?
- [x] Yes (__Insert Feature Release Date Here__)
- [ ] No

---

### CODE REVIEW

File: _docs/_hidden/private_betas/array_of_objects.md [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+        ]
+      }
+    }
+  ]
+}
+```
+{% endtab %}
+{% tab Update %}
+
+Update values for specific objects within an array using the `_merge_objects` parameter and the `$update` operator. Similar to updates to simple [nested custom attribute]({{site.baseurl}}/nested_custom_attribute_support/#api-request-body) objects, this performs a deep merge.
```
Comment (KellieHawks) [inline _docs/_hidden/private_betas/array_of_objects.md:96]: Can you define what a deep merge is here. Couldn't find the definition in the nested custom attribute doc

File: _docs/_hidden/private_betas/array_of_objects.md [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+{% alert warning %}
+You must set `_merge_objects` to true, or your objects will be overwritten. `_merge_objects` is false by default.
+{% endalert %}
+
+{% endtab %}
+{% tab Remove %}
+
+Remove objects from an array using the `$remove` operator in combination with a matching key (`$identifier_key`) and value (`$identifier_value`).
+
+The following example shows removing any object in the pets array that has an `id` with a value of `1`, an `id` with a value of `2`, and a `type` with a value of `dog`. If there are multiple objects with the `type` value of `dog`, all matching objects will be removed.
```
Comment (KellieHawks) [inline _docs/_hidden/private_betas/array_of_objects.md:138]: `pets`

File: _docs/_hidden/private_betas/array_of_objects.md [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+    }
+  ]
+}
+```
+{% endtab %}
+{% endtabs %}
+
+### Liquid Templating
+
+You can use this pets array to personalize a message. The Liquid templating example below shows how to reference the custom attribute object properties saved from the above API request and use them in your messaging.
```
Comment (KellieHawks) [inline _docs/_hidden/private_betas/array_of_objects.md:172]: `pets` here too?

File: _docs/_hidden/private_betas/array_of_objects.md [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+
+When segmenting users based on arrays of objects, a user will qualify for the segment if any object in the array matches the criteria. 
+
+Create a new segment and select **Nested Custom Attribute** as your filter. Then search for and select the name of your array of objects.
+
+![Filter by array of objects][1]
+
+Use dot notation to specify which field in the array of objects you want to use. Start the text field with an empty set of square brackets `[]` to tell Braze that you're looking inside an array of objects. After that, add a period `.`, followed by the name of the field you want to use.
+
+For example, if you want to filter the pets array of objects based on the `type` field, enter `[].type` and choose which type of pet to filter for, such as `snake`.
```
Comment (KellieHawks) [inline _docs/_hidden/private_betas/array_of_objects.md:196]: `pets` here too? Not sure if it's worth formatting it this way for all the instances of "`pets` array". I'd either change the initial one, or codify the text for the rest of the instances.

File: _docs/_hidden/private_betas/array_of_objects.md [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+
+![Filter by pet type equals snake][3]
```
Comment (KellieHawks) [inline _docs/_hidden/private_betas/array_of_objects.md:198]: Just as a nit, can we crop the images a little more to just include the filters?

**Review by KellieHawks — CHANGES_REQUESTED** [review]
Hey Bre! Looks great, just a couple of minor formatting things I wanted to bring up. Let me know if you have any Qs!

File: _docs/_hidden/private_betas/array_of_objects.md (hunk 1 above)
Comment (bre-fitzgerald) [reply _docs/_hidden/private_betas/array_of_objects.md:96]: Good point, added a brief sentence to the nested custom attribute doc.

**Review by KellieHawks — APPROVED** [review]
Looks great! Thanks for those edits!

**Pull request merged**